libs/auth/src/dipped_smart_card_auth.ts

```diff
@@ -336,8 +348,9 @@ export class DippedSmartCardAuth implements DippedSmartCardAuthApi {
         assert(electionHash !== undefined);
         await this.card.program({
           user: { role: 'poll_worker', electionHash },
+          pin: arePollWorkerCardPinsEnabled ? pin : undefined,
         });
-        return undefined;
+        return arePollWorkerCardPinsEnabled ? pin : undefined;
```

Comment:
nit: IMO it's a little safer to put all the logic in a single if block, like
```
    if (arePollWorkerCardPinsEnabled) {
      await this.card.program({
        user: { role: 'poll_worker', electionHash },
        pin,
      });
      return pin;
    }

    await this.card.program({
      user: { role: 'poll_worker', electionHash },
    });
```

but just personal preference

Reply:
Nice, I agree - will update!